### PR TITLE
Delete issues

### PR DESCRIPTION
**Pull request by shawkins**
## Description
Addressing various delete issues from #3982
- adding deleteList methods that provide more information about exactly what has been deleted
- clarified the javadocs of existing delete methods
- have delete() of a list return false when an item can't be deleted
- added these changes to the resourceList interfaces - it was easiest to remove what had already been deprecated

## Type of change

 - [x] Bug fix (non-breaking change which fixes an issue)
 - [x] Feature (non-breaking change which adds functionality)
 - [x] Breaking change (fix or feature that would cause existing functionality to change
 - [ ] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [x] Code contributed by me aligns with current project license: [Apache 2.0](https://www.apache.org/licenses/LICENSE-2.0)
 - [x] I Added [CHANGELOG](https://github.com/fabric8io/kubernetes-client/blob/master/CHANGELOG.md) entry regarding this change
 - [x] I have implemented unit tests to cover my changes
 - [x] I have added/updated the [javadocs](https://www.javadoc.io/doc/io.fabric8/kubernetes-client/latest/index.html) and other [documentation](https://github.com/fabric8io/kubernetes-client/blob/master/doc/CHEATSHEET.md) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=fabric8io_kubernetes-client) report
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

**Review by rohanKanojia — APPROVED** [review]

File: kubernetes-client-api/src/main/java/io/fabric8/kubernetes/client/dsl/MultiDeleteable.java [hunk 1]
```diff
@@ -16,11 +16,47 @@
 
 package io.fabric8.kubernetes.client.dsl;
 
+import java.util.Arrays;
 import java.util.List;
 
-public interface MultiDeleteable<T> {
+public interface MultiDeleteable<T> extends ListDeletable<T> {
```
Comment (XComp) [inline kubernetes-client-api/src/main/java/io/fabric8/kubernetes/client/dsl/MultiDeleteable.java:22]: I'm not convinced that we should extend `ListDeletable` here. It feels that both interfaces serve different purposes: `ListDeletable` has an internal context whereas `MultiDeleteable` expects to get the resources passed as parameters.

Keeping these two separate doesn't harm their usage. Any implementing class can still implement both interfaces. IMHO, interfaces should be a collection of methods that serve the same purpose. That doesn't apply to these two methods in my opinion.

Comment (XComp) [inline kubernetes-client-api/src/main/java/io/fabric8/kubernetes/client/dsl/MultiDeleteable.java:22]:
```suggestion
public interface MultiDeletable<T> extends ListDeletable<T> {
```
I guess, without the `e` is the right version analogous to `ListDeletable`

File: kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/BaseOperation.java [hunk 2]
```diff
@@ -448,8 +449,7 @@ public boolean delete() {
       }
     } else {
       try {
-        deleteList();
-        return true;
+        return delete(list().getItems());
```
Comment (XComp) [inline kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/BaseOperation.java:452]: why not use `deleteList()` to remove redundant code?

File: kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/BaseOperation.java [hunk 3]
```diff
@@ -1009,4 +1003,24 @@ public static URL appendListOptionParams(URL base, ListOptions listOptions) {
     return urlBuilder.build();
   }
 
+  @Override
+  public List<T> deleteList(List<T> items) throws ListDeleteException {
```
Comment (XComp) [inline kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/BaseOperation.java:1007]: There is some redundancy between the `delete(List)` and the `deleteList(List)` here. Maybe, we could align these two in a single private method?
In addition to that, `delete(List)` does a `null` check on the parameter. `deleteList(List)` doesn't.

**Review by XComp — COMMENTED** [review]
Thanks for addressing my comments, @shawkins . I only looked at the list deletion code I described in #3982 . It fixes the overall issue. I have a few comments, though. Please have a look below.

**Comment by shawkins** [comment]
@XComp this pr will change based upon the comments #3982

**Comment by shawkins** [comment]
@manusa here's what this looks like with keeping the boolean return.

**Comment by shawkins** [comment]
After chatting with @manusa we'll update to returning void for now, and capture a new issue for implementing collection delete against the api server (should be a small change) with the eventual goal of processing the return from single item and collection delete to return a list of objects or ids.

**Comment by shawkins** [comment]
The big issue with changing to void rather than boolean is fixing up all of the tests - there are a couple hundred places where we are checking for that response.  I'll separate that off as a different issue.

**Review by manusa — APPROVED** [review]

**Pull request merged**